tests.py

```diff
@@ -1937,6 +2063,68 @@ def test_token_expiration_no_auth(self):
                           [{self.ts.error: "Not authenticated with TimeSync, "
                                            "call self.authenticate() first"}])
 
+    def test_interpret_times(self):
+        """Tests that when a string duration is entered, it is converted to an
+        integer"""
+        param = {
+            "duration": "3h30m",
+            "project": "ganeti-web-manager",
+            "user": "example-user",
+            "activities": ["documenting"],
+            "notes": "Worked on docs",
+            "issue_uri": "https://github.com/",
+            "date_worked": "2014-04-17",
+        }
+
+        self.assertEquals(self.ts.
+                          _TimeSync__interpret_times(param['duration']), 12600)
+
+    def test_interpret_times_with_invalid_str(self):
+        """Tests that when an invalid string duration is entered, duration is
+        set to None"""
+        params = {
+            "duration": "3hh30m",
+            "project": "ganeti-web-manager",
+            "user": "example-user",
+            "activities": ["documenting"],
+            "notes": "Worked on docs",
+            "issue_uri": "https://github.com/",
+            "date_worked": "2014-04-17",
+        }
+
+        self.assertEquals(self.ts.
+                          _TimeSync__interpret_times(params['duration']), None)
+
+    def test_duration_valid(self):
+        """Tests for duration validity"""
+        params = {
+            "duration": 12600,
+            "project": "ganeti-web-manager",
+            "user": "example-user",
+            "activities": ["documenting"],
+            "notes": "Worked on docs",
+            "issue_uri": "https://github.com/",
+            "date_worked": "2014-04-17",
+        }
+
+        self.assertEquals(self.ts.
+                          _TimeSync__check_duration(params['duration']), 12600)
```

Comment:
This is failing tests because

```
AttributeError: 'TimeSync' object has no attribute '_TimeSync__check_duration'
```

Also, I think it would be more readable to do this for indentation:

```
self.assertEquals(
    self.ts._TimeSync__check_duration(params['duration']), 12600)
```

so the object method is on the same line as its object. Same for the other unit tests.